cli/listener/src/main.rs

```diff
@@ -108,33 +100,52 @@ impl IntervalPrices {
     pub fn duration_epochs(&self) -> u64 {
         self.epoch1 - self.epoch0
     }
-    pub fn growth_factor(&self) -> Rational {
-        (self.price1_lamports / self.price0_lamports).expect("Division by 0 cannot happen.")
+    pub fn growth_factor(&self) -> f64 {
+        self.price1_lamports / self.price0_lamports
     }
     pub fn annual_growth_factor(&self) -> f64 {
         let year = chrono::Duration::days(365);
         self.growth_factor()
-            .to_f64()
             .powf(year.num_seconds() as f64 / self.duration_wall_time().num_seconds() as f64)
     }
     pub fn annual_percentage_rate(&self) -> f64 {
         self.annual_growth_factor().mul_add(100.0, -100.0)
     }
 }
 
-pub fn get_apy_for_period(
+impl std::fmt::Display for IntervalPrices {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let duration = self.t1 - self.t0;
+        writeln!(
+            f,
+            "Interval price:\n  From: {} (epoch {})\n  To  : {} (epoch {})\n  Average {} days APY: {}",
+            self.t0,
+            self.epoch0,
+            self.t1,
+            self.epoch1,
+            duration.num_days(),
+            self.annual_percentage_rate()
+        )
+    }
+}
+
+pub fn get_interval_price_for_period(
     tx: rusqlite::Transaction,
-    opts: &Opts,
     from_time: chrono::DateTime<chrono::Utc>,
     to_time: chrono::DateTime<chrono::Utc>,
+    pool: String,
 ) -> rusqlite::Result<Option<IntervalPrices>> {
     let row_map = |row: &Row| {
-        let timestamp_iso8601: String = row.get(1)?;
+        let timestamp: String = row.get(1)?;
+        let timestamp_iso8601 = chrono::DateTime::from_utc(
```

Comment:
We should be able to use [`DateTime::<Utc>::from`](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html#impl-From%3CDateTime%3CFixedOffset%3E%3E), then we don’t have to go through the scary `.naive_utc()`. To be clear, the current code does look correct to me, but I would try to avoid naive datetimes as much as possible.

Reply:
True, didn't see that `DateTime::<Utc>` implemented `FromStr`, thanks!

Reply:
Not `FromStr`, `From`!

Reply:
Ah sure, but we can use [DateTime::<Utc>::from_str](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html#impl-FromStr) to get the string from the data base.